src/main/java/com/aws/greengrass/device/session/SessionCreator.java

```diff
@@ -22,28 +23,23 @@ private SessionCreator() {
         factoryMap = new ConcurrentHashMap<>();
     }
 
-    private static class SessionFactorySingleton {
-        @SuppressWarnings("PMD.AccessorClassGeneration")
-        private static final SessionCreator INSTANCE = new SessionCreator();
-    }
-
     /**
      * Create a client device session.
      *
      * @param credentialType type of credentials provided
-     * @param credentialMap  map of client credentials
+     * @param credential     client credentials
      * @return new session if the client can be authenticated
      * @throws AuthenticationException if the client fails to be authenticated
      */
-    public static Session createSession(String credentialType, Map<String, String> credentialMap)
+    public static Session createSession(String credentialType, Credential credential)
             throws AuthenticationException {
         SessionFactory sessionFactory = SessionFactorySingleton.INSTANCE.factoryMap.get(credentialType);
         if (sessionFactory == null) {
             logger.atWarn().kv("credentialType", credentialType)
-                .log("no registered handler to process device credentials");
+                    .log("no registered handler to process device credentials");
             throw new IllegalArgumentException("unknown credential type");
         }
-        return sessionFactory.createSession(credentialMap);
+        return sessionFactory.createSession(credential);
     }
 
     public static void registerSessionFactory(String credentialType, SessionFactory sessionFactory) {
```

Comment:
Credential registration will break with this change. Please update PR description once tested